File: src/src/validation/Validation.java
package validation;

import java.util.regex.Pattern;
import java.util.regex.Matcher;

public class Validation {

    private static final String PASSWORD_REGEX = "^[a-zA-Z](?=.*[@#$%^&+=])(?=\\S+$).{5,9}$";
    private static final String NAME_REGEX = "^[a-zA-Z]+$";
    private static final String USERNAME_REGEX = "^[a-zA-Z][a-zA-Z0-9]*$";
    private static final String NUMBER_ONLY_REGEX = "^[0-9]+$";


    public static boolean validatePassword(String password) {
        return validatePattern(password, PASSWORD_REGEX);
    }

    public static boolean validateName(String name) {
        return validatePattern(name, NAME_REGEX);
    }

    public static boolean validateUsername(String username) {
        return validatePattern(username, USERNAME_REGEX);
    }
    public static boolean validateNumbers(String number) {
        return validatePattern(number ,NUMBER_ONLY_REGEX);
    }

    private static boolean validatePattern(String input, String regex) {
        Pattern pattern = Pattern.compile(regex);
        Matcher matcher = pattern.matcher(input);
        return matcher.matches();
    }

}

File: src/src/HelpDeskManagement/HelpDeskManagement/HelpDeskManagement/validation/Validation.swift
//
//  Validation.swift
//  HelpDeskManagement
//
//  Created by incubation on 12/11/24.
//

import Foundation

struct Validation {

    private init() { }
    
    private static let PASSWORD_REGEX = #"^[a-zA-Z](?=.*[@#$%^&+=])(?=\S+$).{5,9}$"#
    private static let NAME_REGEX = #"^[a-zA-Z]+$"#
    private static let USERNAME_REGEX = #"^[a-zA-Z][a-zA-Z0-9]*$"#

    static func validatePassword(_ password: String) -> Bool {
        return validatePattern(password, regex: PASSWORD_REGEX)
    }

    static func validateName(_ name: String) -> Bool {
        return validatePattern(name, regex: NAME_REGEX)
    }

    static func validateUsername(_ username: String) -> Bool {
        return validatePattern(username, regex: USERNAME_REGEX)
    }

    private static func validatePattern(_ input: String, regex: String) -> Bool {
        let predicate = NSPredicate(format: "SELF MATCHES %@", regex)
        return predicate.evaluate(with: input)
    }
}
